src/cargo/ops/cargo_add/dependency.rs

```diff
@@ -397,6 +447,21 @@ impl Dependency {
                             table.insert("version", r.into());
                         }
                     }
+                    Some(Source::Workspace(src)) => {
+                        table.set_dotted(true);
+                        table.insert("workspace", src.workspace.into());
+                        if let Some(features) = src.features.as_deref() {
+                            table.insert(
+                                "features",
+                                features.into_iter().collect::<toml_edit::Value>(),
+                            );
+                            table.set_dotted(false);
```

Comment:
Another option is to match for this case specially like we do for `foo = "version"`

Reply:
See
```rust
        let table: toml_edit::Item = match (
            self.optional.unwrap_or(false),
            self.features.as_ref(),
            self.default_features.unwrap_or(true),
            self.source.as_ref(),
            self.registry.as_ref(),
            self.rename.as_ref(),
        ) {
            // Extra short when version flag only
            (
                false,
                None,
                true,
                Some(Source::Registry(RegistrySource { version: v })),
                None,
                None,
            ) => toml_edit::value(v),
            // Other cases are represented as an inline table
            (_, _, _, _, _, _) => {
```
(wish github would let me comment on code outside of the diffs)